core/src/Kokkos_Cuda.hpp

```diff
@@ -245,6 +245,13 @@ class Cuda {
   uint32_t impl_instance_id() const noexcept;
 
  private:
+  friend bool operator==(Cuda const& lhs, Cuda const& rhs) {
+    return lhs.impl_internal_space_instance() ==
+           rhs.impl_internal_space_instance();
```

Comment:
Why don't you use `impl_instance_id()`?

Reply:
Because the pointer to the implementation is what guarantees you get the same scratch allocations and such.
And from that definition one cannot tell how the instance id is being computed.  I would have gone for the actual pointer but I decided to use the `impl_internal_space_instance` member function since it is both available and defined a few lines above